presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ImplementFilteredAggregations.java

```diff
@@ -92,7 +97,17 @@ public Result apply(AggregationNode aggregation, Captures captures, Context cont
                 verify(!mask.isPresent(), "Expected aggregation without mask symbols, see Rule pattern");
                 newAssignments.put(symbol, filter);
                 mask = Optional.of(symbol);
+
+                filterExpressions.add(filter);
```

Comment:
Instead of modifying this loop, I would introduce another `method` that would extract the predicate that you could pushdown or `BooleanLiteral.TRUE` if it is not possible extract such a predicate.